### DISCUSSION TITLE
Internal Pod optimizations for large Pods with many fields (PodsObject, now named Pods\Whatsit)

### DISCUSSION THREAD

**Discussion by sc0ttkclark** [pods-framework/pods#6639] [Ideas]
- [x] Load Pods on demand as needed
- [x] Load only the core Pod information (id, name, label, object, type, storage), load the rest on as needed
- [x] Load only the fields as needed
- [x] Rewrite Traversal tests to use JSON config file and JSON data file, create pods using JSON config and add data using JSON data
- [x] Replace 'options' usage in core
- [x] Give every pod a default group if no groups are defined
- [x] Remove extra PodsMeta caching of registered types
- [x] Map code registered types in PodsMeta to new Pods Whatsit Storage mechanism 
- [ ] Refactor Groups methods in PodsMeta so they use PodsAPI

**Comment by lukebrown** [comment]
+1!

**Comment by pglewis** [comment]
It should be noted that memory usage on 64-bit PHP systems could be as much as double that of 32-bit systems.

**Comment by thewatts** [comment]
This will be great! Especially since our hosting provider (WPEngine) doesn't have a way to currently manage our site running on /latest --- and with such, we're a bit at the mercy of their auto wp-updates. It's something that we'll have to manually revert, as their systems don't allow for "blacklisting" sites.

Thanks so much for your hard work, guys! Pods is the jam!

**Comment by sc0ttkclark** [comment]
PodsObject / PodsObjectField will be handling most of this logic, much more cleanly and optimized than originally planned.

**Comment by zigotica** [comment]
any idea on when will this be public. just crashed a site [MySQL server has gone away] with a pod with around 30 fields and 3 relationships to other much smaller pods. no way to recover, any clue? 
thanx

**Comment by sc0ttkclark** [comment]
Try Pods 2.3.10 (http://pods.io/latest/) and if that doesn't resolve your issue, then try setting this in your wp-config.php:

`define( 'PODS_API_CACHE', false );`

**Comment by zigotica** [comment]
wow, thanx for the fast reply, saved my day!
`define( 'PODS_API_CACHE', false );`
just lets me inside wp-admin now, gonna check if everything is ok
thanx again

**Comment by sc0ttkclark** [comment]
Implemented through PodsObject classes in Pods 3.x branch

**Comment by sc0ttkclark** [comment]
Reopening, to have PodsObject brought over from old 3.0 code into the 2.8 release to go alongside Groups UI #799

**Comment by sc0ttkclark** [comment]
Pretty sure this is all covered but I need to verify the tasks in the tasklist are complete.

**Comment by sc0ttkclark** [comment]
More to be done here on PodsMeta, it's just a bit too tangled up to avoid a complete refactor. Will have to wait a little longer on that part.

**Comment by sc0ttkclark** [comment]
Additional work has been done here for https://github.com/pods-framework/pods/pull/4856 and that seems to just leave groups refactoring.
